src/git_changelog/build.py

```diff
@@ -164,7 +164,7 @@ class Changelog:
     def __init__(  # noqa: WPS231
         self,
         repository: str | Path,
-        provider: ProviderRefParser | None = None,
+        provider: ProviderRefParser | Callable[..., ProviderRefParser] | None = None,
```

Comment:
You meant to accept a provider class, right? I was confused by the callable thing. Maybe you could annotate using `Type[ProviderRefParser]` instead, just like you did somewhere else in the code?